string_map.go

```diff
@@ -0,0 +1,40 @@
+package main
+
+import (
+	"fmt"
+	"strings"
+)
+
+// StringMap is a map of key value pairs.
+type StringMap map[string]string
+
+func (s StringMap) String() string {
+	elements := make([]string, 0, len(s))
+	for k, v := range s {
+		elements = append(elements, fmt.Sprintf("%s=%s", k, v))
+	}
+
+	return strings.Join(elements, ",")
+}
+
+// Set parses a a string of the format: `key=value`.
+func (s StringMap) Set(value string) error {
+	if s == nil {
```

Comment:
interesting that this works with value receiver (map and slice are fine, struct would raise compiler error)